### DISCUSSION TITLE
Advice requested: Shape optimization of transient case.

### DISCUSSION THREAD

**Discussion by ChristianBauerEng** [su2code/SU2#1896] [Q&A]
Hi all,

I'm currently looking into options to perform automatic shape optimization with SU2.
My case is something between the "Unsteady Shape Optimization" and the "Two Way Mixing Channel Optimization" cases. However, due to the nature of the physics involved I require around 120k timesteps to achieve a quasi-steady limit cycle and need to perform at least another 50k TS for meaningful averaging of the objective function.

My idea was to use the FFD-approach of the mixing channel case in a transient configuration. So far I've succeeded in deforming my mesh and set up cases for direct and AD solutions. However, if I understand correctly how the AD works I need a complete restart file for each timestep for the AD solver to work. That would mean that the incurred file overhead is way too prohibitive.

Do you see any smart and efficient way of obtaining gradients and performing optimization?

Cheers,

Christian

**Comment by pcarruscag** [comment]
At that cost you would probably want to create a response surface and run the optimization on that instead.
I do not have experience with them, but there are "gradient informed" response surface methods.
If the number of files is indeed too costly to store, you can obtain gradients in forward mode (one variable at a time). Both forward and adjoint only require you to run the 50k steps for averaging.
You can minimize the file size by only including COORDINATES and SOLUTION in the VOLUME_OUTPUT.
Maybe to obtain the gradients you could also run on a coarser mesh with larger time steps.

**Comment by ChristianBauerEng** [comment]
Thank you for your suggestions.

Unfortunately our parallel file system is horrible at storing many files in a single directory. Unless I can trick SU2 to store all the data in a single file the adjoint simulation is out of the race.

That means that forward-simulations are probably the way to go. However, I doubt that the response will be smooth, which means a response surface may have a hard time representing the physics involved.

Running a single simulation typically takes about 5 days. To speed up the process I'd like to run several simulations in parallel. I'm therefore looking for an algorithm that allows me to seed the design space with a certain number of candidates (parameter combinations) in each design iteration. It should then take these results into account and then refine the optimum search by generating >= 6 new candidates.

Do you have a suggestion on how such a workflow can be achieved? The classical SciPy optimizers seem to generate a single candidate from previous data, while pyDOE methods do not seem to be well-suited for iterative approaches.

**Reply by pcarruscag** [reply]
Depending on how many variables you have, you may run the various forward modes simultaneously to obtain the full gradient.
GA usually uses population sizes larger than 6

**Reply by ChristianBauerEng** [reply]
Good idea, I think that's the way to go.

I'll try to use an FFD box with 5-6 points in horizontal direction and two in vertical. I'll then run two deformation cycles: one in horizontal, one in vertical direction. So the total number of parameters (dV in x and y direction) would be 2*5=10, which is a nice population size to work with for me.

**Comment by bigfooted** [comment]
Is there currently no checkpointing in SU2 to reduce the number of files for unsteady adjoints? Or is that still too much data?

**Reply by ChristianBauerEng** [reply]
If I understood correctly how the AD solver works then it needs the forward-solution restart file for each timestamp.
Since I have to assume that changing an input also changes the limit cycle it may take some time until a new limit-cycle is reached. A number of 50k timestamps is a rather optimistic value in this regard.
I'm told that our file system can only handle a few thousand files in a single directory efficiently. So that doesn't seem to be an option.

**Reply by pcarruscag** [reply]
There are no checkpoints.

**Reply by bigfooted** [reply]
Too bad, I've heard several people talk about this. Googling returns an old project from Erlangen-Nurnberg with code that might work for you, but it's probably is a lot of work to figure out what is going on without guidance. In any case, it's here:
https://blogs.fau.de/konwihr/laufende-projekte/efficient-checkpointing-technique-for-the-unsteady-adjoint-solver-in-su2/
https://github.com/FAU-Aeroacoustics/SU2-Efficient-Output-Control
